InventoryManagement/src/main/java/com/ccs/inventorymanagement/service/ClothingService.java

```diff
@@ -40,11 +41,18 @@ public Mono<Clothing> create(Clothing item) {
 
     @Override
     public Mono<Clothing> update(Clothing item) {
-        return null;
+        return clothingRepository.save(ClothingEntity.from(item)
+                        .toBuilder()
+                        .updated(Instant.now())
+                        .build())
+                .map(Clothing::from);
     }
 
     @Override
-    public Mono<Clothing> delete(UUID id) {
-        return null;
+    public Mono<Void> delete(UUID id) {
+        return clothingRepository.findById(id).map(entity -> entity.toBuilder()
+                        .status(Item.Status.ISSUED)
+                        .build())
+                .flatMap(issuedEntity -> clothingRepository.save(issuedEntity)).then();
```

Comment:
I would imagine the DELETE api marking an item as issued wasn't the goal. If this is going to be a soft delete there should be a more concise Item.Status with a singular meaning. Something like Item.Status.DELETED or a variation of that.

Reply:
Updated